torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1652,6 +1683,16 @@ struct to_ir {
 
       auto unified = unifyTypes(tv->type(), fv->type());
 
+      // If the variable we're looking at is known to be Union[T1, T2],
+      // then it's okay to have one branch return T1 and the other
+      // return T2. (We're not actually going to use `unified` again;
```

Comment:
`We're not actually going to use `unified` again;` seems misleading, it will be used again.